### PR TITLE
[SPARK-30519][CORE] Use spark.executorEnv.HADOOP_USER_NAME to set executor's hadoop user

### PR DESCRIPTION
**Pull request by apeming**
### What changes were proposed in this pull request?

Let executor can specify hadoop user by using spark.executorEnv.HADOOP_USER_NAME

### Why are the changes needed?

We can use HADOOP_USER_NAME to specify hadoop user while submitting a job on driver, but invalid to set executor's hadoop user even by using spark.executorEnv.HADOOP_USER_NAME.

### Does this PR introduce any user-facing change?

No

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ private[spark] class SparkHadoopUtil extends Logging {
   }
 
   def createSparkUser(): UserGroupInformation = {
-    val user = Utils.getCurrentUserName()
+    val user = Option(System.getenv("HADOOP_USER_NAME")).getOrElse(Utils.getCurrentUserName())
```
Comment (seanli-rallyhealth) [inline core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala]: is "HADOOP_USER_NAME" constant defined?

Comment (apeming) [reply core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala]: Yes, it's spec by spark.executorEnv.HADOOP_USER_NAME and all executors will use it

**Comment by jiangxb1987** [comment]
This is a behavior change, and I don't think we should introduce this change by the approach proposed this way.

If both `SPARK_USER` and `HADOOP_USER_NAME` are specified in the env, we should prioritize `SPARK_USER` over `HADOOP_USER_NAME`. Also, I'm hesitate whether we should support `HADOOP_USER_NAME` at all.

**Comment by apeming** [comment]
> This is a behavior change, and I don't think we should introduce this change by the approach proposed this way.
> 
> If both `SPARK_USER` and `HADOOP_USER_NAME` are specified in the env, we should prioritize `SPARK_USER` over `HADOOP_USER_NAME`. Also, I'm hesitate whether we should support `HADOOP_USER_NAME` at all.

But `SPARK_USER` is used for scheduler such as Mesos, Kubernetes and `HADOOP_USER_NAME` is used for Hadoop cluster such HDFS. At Spark on Kubernetes, `SPARK_USER` is not over `HADOOP_USER_NAME`

**Pull request closed without merging**